client/client.go

```diff
@@ -160,25 +170,22 @@ func (c *CfClient) streamConnect() {
 	defer c.mux.RUnlock()
 	c.config.Logger.Info("Registering SSE consumer")
 	sseClient := sse.NewClient(fmt.Sprintf("%s/stream?cluster=%s", c.config.url, c.clusterIdentifier))
-	conn := stream.NewSSEClient(c.sdkKey, c.token, sseClient, c.config.Cache, c.api, c.config.Logger)
-	err := conn.Connect(c.environmentID)
-	if err != nil {
-		c.streamConnected = false
-		return
-	}
 
-	c.streamConnected = true
-	err = conn.OnDisconnect(func() error {
+	streamErr := func() {
+		c.config.Logger.Error("Stream disconnected. Swapping to polling mode")
 		// Wait one minute before moving to polling
 		time.Sleep(1 * time.Minute)
 		c.mux.RLock()
 		defer c.mux.RUnlock()
 		c.streamConnected = false
-		return nil
-	})
-	if err != nil {
-		c.config.Logger.Errorf("error disconnecting the stream, err: %v", err)
 	}
+	conn := stream.NewSSEClient(c.sdkKey, c.token, sseClient, c.config.Cache, c.api, c.config.Logger, streamErr)
+
+	// Connect kicks off a goroutine that attempts to establish a stream connection
+	// while this is happening we set streamConnected to true - if any errors happen
+	// in this process streamConnected will be set back to false by the streamErr function
+	conn.Connect(c.environmentID)
+	c.streamConnected = true
```

Comment:
Does this return any kind of error?   What happens if connect fails here?

Reply:
No it used to in theory but it always just returned hardcoded nil so I removed it from the function signature. `Connect` just kicks off a goroutine that tries to setup the streaming here https://github.com/harness/ff-golang-server-sdk/pull/58/files#diff-7d9bb1457177193134b3f775d399b78e8183e92692012ba5a7af28a6ebd7d340R55 . If something fails in there it'll call the streamError callback function that gets called already if the stream disconnects or can't start that resets us back to polling mode.

Theres a lot of async stuff going on in here that can make it a bit unintuitive to follow at times